### PR TITLE
Separate client/server in different modules

### LINKED ISSUE #112: Support for disabling gRPC server
Hi,

Is there any way to disable a gRPC server to start? In my case, I am using micronaut-grpc for grpc clients only, and it is starting a grpc server unnecessarily.

Shouldn't we have a `grpc.server.enabled` config?

Thanks!

### PR DESCRIPTION
**Pull request by racevedoo**
This is a proposal to fix #112 

Basically it refactors the `grpc-runtime` project in `grpc-client-runtime` and `grpc-server-runtime`, as @graemerocher suggested.

There are some breaking changes here, such as:
- Tracing factories were renamed to `GrpcServerTracingInterceptorFactory` and `GrpcClientTracingInterceptorFactory`. Not sure if that breaks something, but it was worth noting

Some changes are still pending, such as:
- Finding a way to use `GrpcServerChannel.NAME` instead of hardcoding it in client-runtime's `GrpcChannelScope`, as it requires `GrpcEmbeddedServer`
- Placing tests in the correct module(maybe some of them involve both, so I added a testCompile dependency between them)
- Changing most of the documentation

### CODE REVIEW

File: grpc-server-runtime/src/main/resources/META-INF/native-image/io.micronaut.grpc/micronaut-grpc-runtime/native-image.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+Args = --initialize-at-run-time=com.google.protobuf.ExtensionRegistryFactory
```
Comment (graemerocher) [inline grpc-server-runtime/src/main/resources/META-INF/native-image/io.micronaut.grpc/micronaut-grpc-runtime/native-image.properties:1]: Directory needs rename to `micronaut-grpc-server-runtime` or whatever this is appropriate for

File: src/main/docs/guide/client.adoc [hunk 2]
```diff
@@ -25,7 +25,7 @@ Alternatively if you prefer programmatic configuration you can write a `BeanCrea
 .Configuring the NettyChannelBuilder
 [source,java]
 ----
-include::grpc-runtime/src/test/groovy/io/micronaut/grpc/ManagedChannelBuilderListener.java[]
+include::grpc-server-runtime/src/test/groovy/io/micronaut/grpc/ManagedChannelBuilderListener.java[]
```
Comment (graemerocher) [inline src/main/docs/guide/client.adoc:28]: I think there may need to be more documentation updates than just this to refer to the new modules etc.

File: grpc-server-runtime/src/main/resources/META-INF/native-image/io.micronaut.grpc/micronaut-grpc-runtime/native-image.properties (hunk 1 above)
Comment (racevedoo) [reply grpc-server-runtime/src/main/resources/META-INF/native-image/io.micronaut.grpc/micronaut-grpc-runtime/native-image.properties:1]: done!

File: src/main/docs/guide/client.adoc (hunk 2 above)
Comment (racevedoo) [reply src/main/docs/guide/client.adoc:28]: yeah, just updated them

**Comment by graemerocher** [comment]
Thanks for the contribution!

**Pull request merged**